roqollage/src/interface.rs

```diff
@@ -1537,7 +1582,11 @@ pub fn add_gate(
                     render_pragmas,
                 )?;
             }
-            flatten_qubits(circuit_gates, &qubits);
+            flatten_qubits(
+                circuit_gates,
+                &(*qubits.iter().min().unwrap()..=*qubits.iter().max().unwrap())
+                    .collect::<Vec<_>>(),
```

Comment:
This piece of code is reproduced many times. Should it be turned into an inline function?